SimulationAgent/Agent.cs

```diff
@@ -394,6 +404,20 @@ private void TryToStartThreads()
                         this.runningToken.Token));
             }
 
+            this.deviceReplayTask = this.factory.Resolve<IDeviceReplayTask>();
+            this.deviceReplayThread = new Thread(
+                () => this.deviceReplayTask.RunAsync(
+                    this.simulationManagers,
+                    this.deviceReplayActors,
+                    this.runningToken.Token));
+
+            this.deviceReplayTask = this.factory.Resolve<IDeviceReplayTask>();
```

Comment:
This seems identical with the above one.

Reply:
good catch, removed